### ISSUE TITLE
error formatting `stdlib` of Julia

### ISSUE THREAD

**Issue by Moelf** [domluna/JuliaFormatter.jl#644]
```julia

julia> format("./stdlib/", MinimalStyle(); normalize_line_endings = "unix", align_assignment=true)
[ Info: Error in formatting file stdlib/InteractiveUtils/test/runtests.jl
ERROR: ArgumentError: invalid index: nothing of type Nothing
```

**Comment by domluna** [comment]
@Moelf can you post the commit of the Julia repo so I can debug

**Comment by Moelf** [comment]
there's no additional msg
```julia
[ Info: Error in formatting file stdlib/InteractiveUtils/test/runtests.jl
Formatting stdlib/InteractiveUtils/test/runtests.jl
ERROR: ArgumentError: invalid index: nothing of type Nothing
Stacktrace:
  [1] to_index(i::Nothing)
    @ Base ./indices.jl:300
  [2] to_index(A::Vector{JuliaFormatter.FST}, i::Nothing)
    @ Base ./indices.jl:277
  [3] to_indices
    @ ./indices.jl:333 [inlined]
  [4] to_indices
    @ ./indices.jl:325 [inlined]
  [5] getindex
    @ ./abstractarray.jl:1241 [inlined]
  [6] getindex
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/fst.jl:182 [inlined]
  [7] align_binaryopcalls!(fst::JuliaFormatter.FST, op_idxs::Vector{Int64})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/align.jl:124
  [8] align_fst!(fst::JuliaFormatter.FST, opts::JuliaFormatter.Options)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/align.jl:31
  [9] format_text(cst::CSTParser.EXPR, style::MinimalStyle, s::JuliaFormatter.State)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:649
 [10] format_text(text::String, style::MinimalStyle, opts::JuliaFormatter.Options)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:632
 [11] #format_text#238
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:604 [inlined]
 [12] #format_text#237
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:598 [inlined]
 [13] _format_file(filename::String; overwrite::Bool, verbose::Bool, format_markdown::Bool, format_options::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment), Tuple{MinimalStyle, String, Bool}}})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:723
 [14] (::JuliaFormatter.var"#246#249"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}})(formatted_file::Bool, file::String)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:878
 [15] BottomRF
    @ ./reduce.jl:81 [inlined]
 [16] _foldl_impl(op::Base.BottomRF{JuliaFormatter.var"#246#249"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}}}, init::Bool, itr::Vector{String})
    @ Base ./reduce.jl:62
 [17] foldl_impl
    @ ./reduce.jl:48 [inlined]
 [18] mapfoldl_impl
    @ ./reduce.jl:44 [inlined]
 [19] _mapreduce_dim
    @ ./reducedim.jl:362 [inlined]
 [20] #mapreduce#765
    @ ./reducedim.jl:357 [inlined]
 [21] #reduce#767
    @ ./reducedim.jl:406 [inlined]
 [22] (::JuliaFormatter.var"#245#248"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}})(formatted_path::Bool, dir_branch::Tuple{String, Vector{String}, Vector{String}})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:859
 [23] BottomRF
    @ ./reduce.jl:81 [inlined]
 [24] _foldl_impl(op::Base.BottomRF{JuliaFormatter.var"#245#248"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}}}, init::Bool, itr::Channel{Tuple{String, Vector{String}, Vector{String}}})
    @ Base ./reduce.jl:62
 [25] foldl_impl(op::Base.BottomRF{JuliaFormatter.var"#245#248"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}}}, nt::Bool, itr::Channel{Tuple{String, Vector{String}, Vector{String}}})
    @ Base ./reduce.jl:48
 [26] mapfoldl_impl
    @ ./reduce.jl:44 [inlined]
 [27] #mapfoldl#259
    @ ./reduce.jl:170 [inlined]
 [28] #mapreduce#263
    @ ./reduce.jl:302 [inlined]
 [29] #reduce#265
    @ ./reduce.jl:483 [inlined]
 [30] format(paths::Tuple{String}; options::Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:856
 [31] #format#250
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:889 [inlined]
 [32] #format#251
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:894 [inlined]
 [33] top-level scope
    @ REPL[3]:1

caused by: ArgumentError: invalid index: nothing of type Nothing
Stacktrace:
  [1] to_index(i::Nothing)
    @ Base ./indices.jl:300
  [2] to_index(A::Vector{JuliaFormatter.FST}, i::Nothing)
    @ Base ./indices.jl:277
  [3] to_indices
    @ ./indices.jl:333 [inlined]
  [4] to_indices
    @ ./indices.jl:325 [inlined]
  [5] getindex
    @ ./abstractarray.jl:1241 [inlined]
  [6] getindex
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/fst.jl:182 [inlined]
  [7] align_binaryopcalls!(fst::JuliaFormatter.FST, op_idxs::Vector{Int64})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/align.jl:124
  [8] align_fst!(fst::JuliaFormatter.FST, opts::JuliaFormatter.Options)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/align.jl:31
  [9] format_text(cst::CSTParser.EXPR, style::MinimalStyle, s::JuliaFormatter.State)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:649
 [10] format_text(text::String, style::MinimalStyle, opts::JuliaFormatter.Options)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:632
 [11] #format_text#238
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:604 [inlined]
 [12] #format_text#237
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:598 [inlined]
 [13] _format_file(filename::String; overwrite::Bool, verbose::Bool, format_markdown::Bool, format_options::Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment), Tuple{MinimalStyle, String, Bool}}})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:723
 [14] (::JuliaFormatter.var"#246#249"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}})(formatted_file::Bool, file::String)
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:873
 [15] BottomRF
    @ ./reduce.jl:81 [inlined]
 [16] _foldl_impl(op::Base.BottomRF{JuliaFormatter.var"#246#249"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}}}, init::Bool, itr::Vector{String})
    @ Base ./reduce.jl:62
 [17] foldl_impl
    @ ./reduce.jl:48 [inlined]
 [18] mapfoldl_impl
    @ ./reduce.jl:44 [inlined]
 [19] _mapreduce_dim
    @ ./reducedim.jl:362 [inlined]
 [20] #mapreduce#765
    @ ./reducedim.jl:357 [inlined]
 [21] #reduce#767
    @ ./reducedim.jl:406 [inlined]
 [22] (::JuliaFormatter.var"#245#248"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}})(formatted_path::Bool, dir_branch::Tuple{String, Vector{String}, Vector{String}})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:859
 [23] BottomRF
    @ ./reduce.jl:81 [inlined]
 [24] _foldl_impl(op::Base.BottomRF{JuliaFormatter.var"#245#248"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}}}, init::Bool, itr::Channel{Tuple{String, Vector{String}, Vector{String}}})
    @ Base ./reduce.jl:62
 [25] foldl_impl(op::Base.BottomRF{JuliaFormatter.var"#245#248"{Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}}}}, nt::Bool, itr::Channel{Tuple{String, Vector{String}, Vector{String}}})
    @ Base ./reduce.jl:48
 [26] mapfoldl_impl
    @ ./reduce.jl:44 [inlined]
 [27] #mapfoldl#259
    @ ./reduce.jl:170 [inlined]
 [28] #mapreduce#263
    @ ./reduce.jl:302 [inlined]
 [29] #reduce#265
    @ ./reduce.jl:483 [inlined]
 [30] format(paths::Tuple{String}; options::Base.Pairs{Symbol, Any, NTuple{4, Symbol}, NamedTuple{(:style, :normalize_line_endings, :align_assignment, :verbose), Tuple{MinimalStyle, String, Bool, Bool}}})
    @ JuliaFormatter ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:856
 [31] #format#250
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:889 [inlined]
 [32] #format#251
    @ ~/.julia/packages/JuliaFormatter/xKNsr/src/JuliaFormatter.jl:894 [inlined]
 [33] top-level scope
    @ REPL[3]:1
```

**Comment by domluna** [comment]
yeah that's fine, it's mainly to just track down the file that's causing issues

**Comment by Moelf** [comment]
right, that information is already available though, there's a `@info` telling us which file is the offender

**Comment by bramtayl** [comment]
A good start might be to add a more informative error if `binop_idx` is nothing: https://github.com/domluna/JuliaFormatter.jl/blob/6c0f94468eaef63405fbcbb11943314a164262d6/src/align.jl#L123

**Comment by domluna** [comment]
formats fine if binop_idx === nothing is skipped

Seems to be something to do with a macroblock for this particular error

**Comment by domluna** [comment]
ok I have a PR which fixes the `align_assignment` issue #646
